python/pip_install/extract_wheels/lib/bazel.py

```diff
@@ -139,12 +139,10 @@ def generate_build_file_contents(
     there may be no Python sources whatsoever (e.g. packages written in Cython: like `pymssql`).
     """
 
-    # `dist-info` contains non-determinisitc files which can change any time
-    # the repository rules run. Below is a list of known patterns to these
-    # files. However, not all files should be ignored as certain packages
-    # require things like `top_level.txt`.
     dist_info_ignores = [
-        "**/*.dist-info/METADATA",
+        # RECORD is known to contain sha256 checksums of files which might include the checksums
```

Comment:
Are there known examples this happens in? Some import lib functions fail still because they use data from RECORD, the files() method on a distribution for instance.

Reply:
I too have been wondering why we're doing so much non-standard unpacking and exclusion of files from wheel packages. 

I'm trying to land a change that uses `installer` to bring some PEP standards into this rule set. See: https://github.com/bazelbuild/rules_python/pull/715

I think we need a better way to handle non-determinism. I think we need to either make it opt-in somehow, or direct people in how to use a "wheel only" (using `--only-binary=:all:`). If you do that, then the only non-determinism in generated `.pyc` files, which can be excluded/included via some parameters.